src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -103,6 +105,51 @@ export const getThorTradeQuote = async (
     affiliateBps: requestedAffiliateBps,
   })
 
+  const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+  const maybePoolsResponse = await thorService.get<ThornodePoolResponse[]>(
+    `${daemonUrl}/lcd/thorchain/pools`,
+  )
+
+  if (maybePoolsResponse.isErr()) return Err(maybePoolsResponse.unwrapErr())
+
+  const { data: poolsResponse } = maybePoolsResponse.unwrap()
+
+  const buyPoolId = assetIdToPoolAssetId({ assetId: buyAsset.assetId })
+  const sellPoolId = assetIdToPoolAssetId({ assetId: sellAsset.assetId })
+  const sellAssetPool = poolsResponse.find(pool => pool.asset === sellPoolId)
+  const buyAssetPool = poolsResponse.find(pool => pool.asset === buyPoolId)
+
+  if (!sellAssetPool)
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote]: Pool not found for sell asset ${sellAsset.assetId}`,
+        code: SwapErrorType.POOL_NOT_FOUND,
+        details: { sellAssetId: sellAsset.assetId, buyAssetId: buyAsset.assetId },
+      }),
+    )
+
+  if (!buyAssetPool)
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote]: Pool not found for buy asset ${buyAsset.assetId}`,
+        code: SwapErrorType.POOL_NOT_FOUND,
+        details: { sellAssetId: sellAsset.assetId, buyAsset: buyAsset.assetId },
+      }),
+    )
+
+  const sellAssetDepthBps = sellAssetPool.derived_depth_bps
+  const buyAssetDepthBps = buyAssetPool.derived_depth_bps
+  const swapDepthBps = bn(sellAssetDepthBps).plus(buyAssetDepthBps).div(2)
```

Comment:
Note, this is possibly wrong, waiting on the answer from 9R on this one.
But as noted in the PR description, this will make it a 10 blocks interval at *worst* regardless of the math used, which isn't worse than now.